### PR TITLE
fix: use `ServiceOperationError` and remove context

### PR DESCRIPTION
**Pull request by erik-slovak**
Dependencies:
- [x] https://github.com/technology-studio/service-prop/pull/587
- [x] https://github.com/technology-studio/service-error-handler-react-peer/pull/687
- [x] https://github.com/technology-studio/service-error-handler-react/pull/718

BREAKING CHANGE:
Renamed `exception` to `error` on `useServiceQuery`
Requires packages:
- `@txo/service-prop@^3.0.0`
- `@txo-peer-dep/service-error-handler-react@^2.0.0`
- `@txo/service-error-handler-react@^2.0.0`

### CODE REVIEW

File: src/Hooks/UseServiceQuery.ts [hunk 1]
```diff
@@ -118,32 +118,31 @@ export const useServiceQuery = <
...
       })
       return exception
     }
     return null
-  }, [context, memoizedQuery, queryDocument])
+  }, [context, memoizedQuery.error, queryDocument])
   useLayoutEffect(() => {
-    if ((exception != null) && (shownExceptionListRef.current.find(shownException => (
+    if ((exception != null) && (reportedOperationErrorListRef.current.find(shownException => (
       isServiceErrorListEqual(shownException.serviceErrorList, exception.serviceErrorList)
```
Comment (rostislav-simonik) [inline src/Hooks/UseServiceQuery.ts:132]: this is not needed

File: src/Hooks/UseServiceQuery.ts [hunk 2]
```diff
@@ -118,32 +107,29 @@ export const useServiceQuery = <
...
+    if ((exception != null) && (reportedOperationErrorListRef.current.includes(exception))) {
+      reportError(exception)
+      reportedOperationErrorListRef.current.push(exception)
     }
-  }, [addServiceErrorException, context, exception, memoizedVariables, queryDocument, removeServiceErrorException])
+  }, [context, exception, memoizedVariables, queryDocument])
 
   const promiselessRefetch = useCallback((...args: Parameters<typeof memoizedQuery.refetch>) => {
+    reportedOperationErrorListRef.current = []
     asyncToCallback(memoizedQuery.refetch(...args))
```
Comment (rostislav-simonik) [inline src/Hooks/UseServiceQuery.ts:128]: we need to check if failed refetch includes the error in query result

File: src/Hooks/UseServiceQuery.ts [hunk 3]
```diff
@@ -118,32 +107,29 @@ export const useServiceQuery = <
...
   }, [memoizedQuery])
 
   const fetchMore: QueryResult<DATA>['fetchMore'] = useCallback(async (...args) => {
+    reportedOperationErrorListRef.current = []
     setFetchMoreFetching(true)
     return (
       await memoizedQuery.fetchMore(...args)
```
Comment (rostislav-simonik) [inline src/Hooks/UseServiceQuery.ts:135]: we need to check if failed fetchMore includes the error in query result

File: src/Hooks/UseServiceQuery.ts (hunk 2 above)
Comment (erik-slovak) [reply src/Hooks/UseServiceQuery.ts:128]: Error is added to `query.exception` when using refetch.

File: src/Hooks/UseServiceQuery.ts (hunk 3 above)
Comment (erik-slovak) [reply src/Hooks/UseServiceQuery.ts:135]: `fetchMore` doesn't change `error`, so the `reportError` is probably needed.

Comment (rostislav-simonik) [reply src/Hooks/UseServiceQuery.ts:135]: not reportError, but just throwwing the erorr.

File: src/Hooks/UseServiceQuery.ts [hunk 4]
```diff
@@ -118,32 +107,29 @@ export const useServiceQuery = <
         context,
         operationName,
       })
-      const exception = new ServiceErrorException({
+      const exception = new ServiceOperationError({
```
Comment (rostislav-simonik) [inline src/Hooks/UseServiceQuery.ts:110]:
```suggestion
      const error = new ServiceOperationError({
```

File: src/Hooks/UseServiceQuery.ts [hunk 5]
```diff
@@ -118,32 +107,29 @@ export const useServiceQuery = <
...
         serviceErrorList: errorList,
```
Comment (rostislav-simonik) [inline src/Hooks/UseServiceQuery.ts:111]:
```suggestion
        serviceErrorList,
```

File: src/Hooks/UseServiceQuery.ts [hunk 6]
```diff
@@ -153,19 +139,18 @@ export const useServiceQuery = <
             context,
             operationName,
           })
-          const exception = new ServiceErrorException({
+          const exception = new ServiceOperationError({
```
Comment (rostislav-simonik) [inline src/Hooks/UseServiceQuery.ts:142]:
```suggestion
          const serviceOperationError = new ServiceOperationError({
```

File: package.json [hunk 7]
```diff
@@ -16,47 +16,47 @@
...
     "build:watch": "yarn tsc --watch",
-    "test": "jest",
-    "test:watch": "concurrently \"yarn build:watch\" \"jest --watch\"",
-    "coverage": "jest --coverage && open coverage/lcov-report/index.html || xdg-open coverage/lcov-report/index.html",
+    "build": "yarn build:clean && yarn build:lib",
     "compare-boilerplate-version": "./scripts/compare-boilerplate-version.sh",
-    "lint": "eslint --max-warnings 0 .",
-    "fixcode": "eslint . --fix",
+    "coverage": "jest --coverage && open coverage/lcov-report/index.html || xdg-open coverage/lcov-report/index.html",
     "git-hook": "yarn -s lint-staged",
```
Comment (rostislav-simonik) [inline package.json:25]: is this still needed ?

**Review by rostislav-simonik — APPROVED** [review]

**Pull request merged**